Recommend master branch protection for GitHub host repositories

pmcnr-hx opened the pull request:
GitHub now supports server side protection of branches. Our recommended default setting is to enable this for all our repositories hosted on GitHub.
- [x] +1
- [x] +2

pmcnr-hx committed:
Add section about master branch protection.

Branches can now be protected server side when hosted on GitHub. This is
a good thing because it can easily prevent accidents. Add section to Git
best practices recommending it being enabled for all our repositories.

hx-markterry commented:
ok with me :full_moon_with_face:

lukefrake commented:
Love this!

In SB we have staging branch as well, I think it would make sense to say '_master_ and _staging_' branches protected, to be inclusive of us.

pmcnr-hx commented:
Thanks for the heads up @lukefrake: will update accordingly.

pmcnr-hx committed:
Mention staging branches and clarify merges.

Mention staging branches, because SB also uses those, and clarify that
enabling status checks for merges is optional.

hx-markterry commented:
we can just say "non feature branches" to be inclusive for all projects as the mobile app has the "edge" branch as well as "master"

pmcnr-hx commented:
We can indeed. Inclusive wording. Will update.

pmcnr-hx committed:
Reworded to include all non-feature branches.
